frontend/src/logic/requestlogic.ts

```diff
@@ -1,296 +1,319 @@
     })
-  );
-  return requests.filter(
-    (value: Request) => value.task.status === "OPEN"
-  ).sort((prev, curr) => sortByReverseOrder(prev.task.expectedDeliveryTime, curr.task.expectedDeliveryTime));
-};
+
+    const requests: Request[] = []
+    await Promise.all(
+        temp.map(async (requestSnapshot) => {
+            try {
+                const request = await requestConverter.fromFirestore(
+                    requestSnapshot
+                )
+                requests.push(request)
+            } catch (e) {
+                return console.error(
+                    `Encountered error while retrieving request: ${e.message}`
+                )
+            }
+        })
+    )
+    return requests
+        .filter((value: Request) => value.task.status === 'OPEN')
+        .sort((prev, curr) =>
+            sortByReverseOrder(
+                prev.task.expectedDeliveryTime,
+                curr.task.expectedDeliveryTime
+            )
+        )
+}
 
 type CurrentRequests = {
-  open: Request[],
-  pending: Request[]
+    open: Request[]
+    pending: Request[]
 }
 /**
- * Gets all current requests that are associated with `uid`. 
+ * Gets all current requests that are associated with `uid`.
  * The requests are categorised by `OPEN` and `PENDING`.
  * @throws Error if `uid` is empty.
  */
 export const getCurrentRequests: (
-  uid: string
+    uid: string
 ) => Promise<CurrentRequests> = async (uid: string) => {
-  try {
-    ensureNonEmpty(uid);
-  } catch (e) {
-    throw new Error("Unable to get requests when uid is empty");
-  }
-  
-  const requestRef = db
-    .collection(COLLECTION_REQUESTS)
-    .where("uid", "==", uid)
-    .where("status", "in", [Status.OPEN, Status.PENDING]);
-  const requestQuerySnapshot = await requestRef.get();
-
-  const requests: firebase.firestore.QueryDocumentSnapshot<
-    firebase.firestore.DocumentData
-  >[] = [];
-  requestQuerySnapshot.forEach((requestSnapshot) => {
-    requests.push(requestSnapshot);
-  });
-
-  const results: Request[] = [];
-  await Promise.all(
-    requests.map(async (requestSnapshot) => {
-      try {
-        const request = await requestConverter.fromFirestore(requestSnapshot);
-        results.push(request);
-      } catch (e) {
-        return console.error(
-          `Encountered error while retrieving request: ${e.message}`
-        );
-      }
-    })
-  );
-
-  const categorisedResults: CurrentRequests = {
-    open: [],
-    pending: []
-  }
+    try {
+        ensureNonEmpty(uid)
+    } catch (e) {
+        throw new Error('Unable to get requests when uid is empty')
+    }
 
-  results.sort((prev, curr) => sortByReverseOrder(prev.task.expectedDeliveryTime, curr.task.expectedDeliveryTime))
-    .forEach(request => {
-      const status: Status = request.task.status;
+    const requestRef = db
+        .collection(COLLECTION_REQUESTS)
+        .where('uid', '==', uid)
+    //.where('status', 'in', [Status.OPEN, Status.PENDING])
```

Comment:
@pohlinwei This status was a typo right? i commented it out

Reply:
Why do you think it is a typo? 😅